flocker/control/_model.py

```diff
@@ -609,6 +609,47 @@ def expire(self, now):
         return updated
 
 
+class DatasetAlreadyOwned(Exception):
+    """
+    There is already a blockdevice owned by the given dataset, when trying to
+    record ownership of a differnt blockdevice.
+    """
+
+
+class BlockDeviceOwnership(CheckedPMap):
+    """
+    Persistent mapping of blockdevices to datasets.
+    """
+    __key_type__ = UUID
+    __value_type__ = unicode
+
+    def record_ownership(self, dataset_id, blockdevice_id):
+        """
+        Record that blockdevice_id is the relevant one for given dataset_id.
+
+        Once a record is made no other entry can overwrite the existing
+        one; the relationship is hardcoded and permanent. XXX this may
+        interact badly with deletion of dataset where dataset_id is
+        auto-generated from name, e.g. flocker-deploy or Docker
+        plugin. That is pre-existing issue, though.
+        """
+        current_blockdevice_id = self.get(blockdevice_id)
+        if current_blockdevice_id not in (None, blockdevice_id):
+            raise DatasetAlreadyOwned()
+        return self.set(dataset_id, blockdevice_id)
+
+
+class PersistentState(PClass):
```

Comment:
Now I've found the reference to ConfigurationPersistenceService, so it
seems like we frequently serialize the configuration to this service,
and that will include this new mapping.

Presumably there's a window of potential data loss with this arrangement,
but that's the design we have chosen.